packages/settings-library/src/settings_library/basic_types.py

```diff
@@ -55,7 +51,6 @@ class BuildTargetEnum(str, Enum):
 
 # non-empty bounded string used as identifier
 # e.g. "123" or "name_123" or "fa327c73-52d8-462a-9267-84eeaf0f90e3" but NOT ""
-class IDStr(ConstrainedStr):
-    strip_whitespace = True
-    min_length = 1
-    max_length = 50
+IDStr: TypeAlias = Annotated[
+    str, StringConstraints(strip_whitespace=True, min_length=1, max_length=50)
+]
```

Comment:
maybe use your constraint string class? Does validation still run when I run: `IDStr("a"*100)` this should raise an error. If it raises and error I'm fine with the current.